app/Http/Controllers/ProfileController.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Storage;
+use App\Models\User;
+use App\Models\Profile;
+use App\Models\Level;
+
+class ProfileController extends Controller
+{
+    const LOCAL_STORAGE_FOLDER = 'public/avatars/';
+    private $user;
+
+    public function __construct(User $user, Level $level){
+        $this->user = $user;
+        $this->level = $level;
+    }
+
+    public function show($id){
+        $user = $this->user->findOrFail($id);
+        return view('users.profile.show')
+        ->with('user', $user);
+    }
+
+    public function edit(){
+        $user = $this->user->findOrFail(Auth::user()->id);
+
+        $all_levels = $this->level->all();
+
+        return view('users.profile.edit')
+        ->with('user', $user)
+        ->with('all_levels', $all_levels);
+    }
+
+    public function update(Request $request){
+        $user = $this->user->findOrFail(Auth::user()->id);
+
+        $all_levels = $this->level->all();
+
+        $request->validate([
+            'avatar' => 'max:1048|mimes:jpeg,jpg,png,gif',
+            'level_id' => 'required',
+            'user_name' => 'required|max:50',
+            'comment' => 'max:100',
```

Comment:
Pls make this atleast 250 character

Reply:
Modified as follows.
 
 $request->validate([
            'avatar' => 'max:1048|mimes:jpeg,jpg,png,gif',
            'level_id' => 'required',
            'user_name' => 'required|max:250',
            'comment' => 'max:250',
            'email' => 'required|max:250'
        ]);